doc/source/structures/misc/string.rst

```diff
@@ -326,6 +329,48 @@ Structure
 
     returns a new string with no leading whitespace
 
+.. method:: String:TONUMBER(defaultIfError)
+
+    :parameter defaultIfError: (optional argument) :struct:`Scalar` to return as a default value if the string format is in error.
+    :return: :struct:`Scalar`
+
+    Returns the numeric version of the string, as a number that can be used
+    for mathematics or anywhere a :struct:`Scalar` is expected.  If the
+    string is not in a format that kOS is able to convert into a number, then
+    the value ``defaultIfError`` is returned instead.  You can use this to
+    either select a sane default, or to deliberately select a value you
+    never expect to get in normal circumstances so you can use it as a
+    test to see if the string was formatted well.
+
+    The argument ``defaultIfError`` is optional.  If it is left off, then
+    when there is a problem in the format of the string, you will get
+    an error that stops the script instead of returning a value.
+
+    The valid understood format allows an optional leading sign,
+    a decimal point with fractional part, and scientific notation
+    using "e" as in "1.23e3" for "1230" or "1.23e-3" for "0.00123".
```

Comment:
Well, TONUMBER can be changed to strip all spaces not just leading and trailing (so "1.23 e - 3" would work, but that would also make "1   2" get parsed as 12 instead of as an error, if you're okay with that.